docs/_guides/working-with-events.md

```diff
@@ -0,0 +1,109 @@
+
+```
+           NAME          | SOURCE TYPE |    EVENT TYPE    | FUNCTION NAME | STATUS |         CREATED DATE
+---------------------------------------------------------------------------------------------------------
+  complete-cicada-410962 | *           | vm.being.created | myFunctio     | READY  | Fri Dec 31 17:18:54 PST -0001
+```
+
+The above subscription will cause dispatch to execute function `myFunction` for every event of type `vm.being.created`. 
+Note the asterisk in  `SOURCE TYPE` column. This is default value and say "match all source Types".
+What is source type? If you create event driver of `vcenter` type, `vcenter` is your source type. 
+
+You can also specify a name for your subscription using `--name` parameter. if you don't, a random, human-readable name will be created.  
+
+### Event driver event types
+
+To find out the list of event types produced by built-in event drivers, see [Built-in Event Drivers](built-in-event-drivers.md).
+
+
+## Emitting events via CLI or API
+
+Event driver is a powerful concept, but there are cases when you want to emit an event by your self or through some external application.
+Dispatch exposes dedicated API to emit events.
+
+To emit an event using CLI, you can use `dispatch emit` subcommand.
+
+In a following example, We emit an event of type `my.event`:
+
+```
+dispatch emit my.event --data '{ "example": "payload"}'
+```  
+
+Events in Dispatch follow [Cloud Events specification](https://github.com/cloudevents/spec/blob/460a90d7a69f6257246487e37746797aa2ae919f/spec.md).
+There are certain attributes in Cloud Events that are mandatory. when emitting them through CLI, CLI will pick reasonable defaults.
+You can customize those defaults using CLI attributes:
+
+* `--event-namespace` - Event namespace. Defaults to `dispatchframework.io` 
+* `--source-type` - Event source type. Defaults to `dispatch`
+* `--source-id` - Event source ID. Defaults to `dispatch`
+* `--event-id` - Event ID, should be unique within the scope of producer. Defaults to generated UUIDv4. 
+  NOTE: If source ID is `dispatch`, event-id will ALWAYS be overwritten by Dispatch and set to random UUIDv4. 
+
+You can also provide following optional details about the emitted event:
+* `--event-type-version` - Version of the event, specific to the source. Defaults to empty string.
+* `--schema-url` - URL to schema describing event data. Defaults to empty string.
+* `--content-type` - Content Type of optional data payload. Defaults to `application/json` when using `--data`, or automatically
+detected type when using `--data-from-file`. If specified, always takes precedence over detected type.
+
+You can provide data payload for Event using `--data` parameter. `--data-from-file` will read the content from specified
+file path. Use `--binary` switch to automatically encode the data using base64. If dispatch CLI detects binary type,
+the content will automatically be base64-encoded.
+
+TODO: Add API example.
```

Comment:
Are you going to do this now? Should we take this out before we publish the PR? Same thing for the TODO a few lines down.